scripts/surgical_breakdown_executor.py

```
@@ -0,0 +1,272 @@
+        
+        try:
+            # Check file count
+            result = subprocess.run(
+                ["git", "diff", "--name-only", "main", branch_name],
+                capture_output=True, text=True, check=True
+            )
+            changed_files = [f for f in result.stdout.strip().split('\n') if f]
+            file_count = len(changed_files)
+            
+            # Check line count
+            result = subprocess.run(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)